cedar-policy/src/frontend/is_authorized.rs

```diff
@@ -143,17 +144,17 @@ enum AuthorizationAnswer {
 #[serde_as]
 #[derive(Debug, Serialize, Deserialize)]
 struct AuthorizationCall {
-    principal: Option<serde_json::Value>,
-    action: serde_json::Value,
-    resource: Option<serde_json::Value>,
+    principal: Option<JsonValueWithNoDuplicateKeys>,
+    action: JsonValueWithNoDuplicateKeys,
+    resource: Option<JsonValueWithNoDuplicateKeys>,
     #[serde_as(as = "MapPreventDuplicates<_, _>")]
-    context: HashMap<String, serde_json::Value>,
+    context: HashMap<String, JsonValueWithNoDuplicateKeys>,
     /// Optional schema in JSON format.
     /// If present, this will inform the parsing: for instance, it will allow
     /// `__entity` and `__extn` escapes to be implicit, and it will error if
     /// attributes have the wrong types (e.g., string instead of integer).
     #[serde(rename = "schema")]
-    schema: Option<serde_json::Value>,
```

Comment:
Schema parsing _should_ already be forbidding duplicate keys everywhere, so this should just move the error message around a bit. I don't see anything wrong with that, and consistently using the no-duplicates variant is nice, but maybe think about if the new error is substantially better/worse than before.

Reply:
I don't think that's the case.  `serde_json::Value` cannot contain duplicate keys, so any schema parsing that accepts `serde_json::Value` won't ever see the duplicate keys -- they would already be silently dropped (last-value-wins) during whatever process created the `serde_json::Value`.